### ISSUE TITLE
BH Install Checker (BICH) - wrong path detected, Linux

### ISSUE THREAD

**Issue by JBHann** [Gameslinx/BeyondHomePlanetMod#11]
[LOG 22:42:33.717] [BICH] GameData directory is /mnt/mydrive/myGames/steamapps/common/Kerbal Space ProgGameData/

BICH is cutting off too much before appending "GameData/" . For some reason it is deleting the "ram/" and getting the wrong path. The source code shows BICH grabs the path variable and deletes the last 12 characters. Maybe a fixed 12 characters is correct for Windows but incorrect for Linux?

The symptoms I experience in game Kopernicus 1.10 (DEV version) loads the planets correctly, but they are all missing textures (shows as pure magenta) when viewing the space center or when nearby with a ship. Coloring/textures appear correct when viewing the map or tracking center. This could be some other mod causing this but the only mods I have installed are dependencies and the log file and BICH seem to point to something within the Beyond Home compiled .dlls. 

I have included the modulemanager output from the KSP.log so you can see what path BICH is working with.

Extracts from KSP.log follow:
[BICH_log_excerpt.txt](https://github.com/Gameslinx/BeyondHomePlanetMod/files/5406308/BICH_log_excerpt.txt)

**Comment by DeuxVis** [comment]
Probably related to that line : https://github.com/Gameslinx/BeyondHomePlanetMod/blob/f17fd5f67bbbb5f36a51eccc728f3a16452474bc/GameData/BeyondHome/BeyondHome_InstallChecker-SOURCE.cs#L30

I have that "everything purple" problem in Linux too, so thanks for the finding.

**Comment by JBHann** [comment]
Yes, I think the installation checker log is related to line 30 there. I assume there is corresponding logic in Beyond Home somewhere as well, but I don't see other source code. This may mean the issue for symptoms beyond the logging done by BICH is in Kopernicus somewhere, but in any case fixing the installation checker seems reasonable.
Presumably it could be changed to an adaptive loop like:

string path = Application.dataPath.Remove(Application.dataPath.Length - 6, 6) + "GameData/";
while (path[last] != '/' AND path[last] != '/') {
path = path.Remove(path.Length -1, 1);
}
path = path + "GameData/";
///// Apologies for the pseudo-code and syntax. I don't know what language this source is written in and I am guessing on the expression to remove the last character of the string and don't know the syntax to read/check the last character of the string "path[last]". Replace logic with appropriate syntax.

**Comment by JBHann** [comment]
I did some testing. Apparently the magenta textures problem is directly related to the shader, and Parallax specifically.
Evidence:
* Deleting the Parallax folder entirely restores the planet textures and they are no longer magenta. Planetary rings are still magenta.
* The configuration file of Gateway.cfg for the rings has the line "useNewShader = true". If this line is removed the display of the ring is still broken, but it is no longer magenta. It becomes a faint transparent black set of rings.

For now a work-around seems to be to manually delete the Parallax folder after installing via CKAN. The Beyond Home planetary system will still install via Kopernicus.

According to this new testing, the main issue appears to be with Parallax and not Beyond Home. Regarding the install checker, I note the Parallax source uses a different determination of path:
string filePath = Path.Combine(KSPUtil.ApplicationRootPath + "GameData/" + "Parallax/Shaders/Parallax");
